addons/stock/models/stock_storage_category.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+
+
+class StorageCategory(models.Model):
+    _name = 'stock.storage.category'
+    _description = "Storage Category"
+    _order = "name"
+
+    name = fields.Char('Storage Category', required=True)
+    max_weight = fields.Float('Max Weight', digits='Stock Weight')
+    capacity_ids = fields.One2many('stock.storage.category.capacity', 'storage_category_id')
+    product_capacity_ids = fields.One2many('stock.storage.category.capacity', compute="_compute_storage_capacity_ids", inverse="_set_storage_capacity_ids")
+    package_capacity_ids = fields.One2many('stock.storage.category.capacity', compute="_compute_storage_capacity_ids", inverse="_set_storage_capacity_ids")
+    allow_new_product = fields.Selection([
+        ('empty', 'If the location is empty'),
+        ('same', 'If all products are same'),
+        ('mixed', 'Allow mixed products')], default='mixed', required=True)
+    location_ids = fields.One2many('stock.location', 'storage_category_id')
+    company_id = fields.Many2one('res.company', 'Company')
+
+    _sql_constraints = [
+        ('positive_max_weight', 'CHECK(max_weight >= 0)', 'Max weight should be a positive number.'),
+    ]
+
+    @api.depends('capacity_ids')
```

Comment:
I don't understand this change. I don't think we need to trigger a recompute when a line changed it's product_id, right?

Reply:
If you put the product_id of one of `capacity_ids` into False (you remove the product_id in a line). But it cannot happen because it is required on the form, then it is correct.